core/presentation/designsystem/src/main/java/com/pronoidsoftware/core/presentation/designsystem/components/TaskyDatePicker.kt

```diff
@@ -0,0 +1,79 @@
+@file:OptIn(ExperimentalMaterial3Api::class)
+
+package com.pronoidsoftware.core.presentation.designsystem.components
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Row
+import androidx.compose.material3.DatePicker
+import androidx.compose.material3.DatePickerDialog
+import androidx.compose.material3.DatePickerState
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import com.pronoidsoftware.core.domain.util.toLocalDateFromUTC
+import com.pronoidsoftware.core.domain.util.toMillis
+import com.pronoidsoftware.core.domain.util.today
+import com.pronoidsoftware.core.presentation.designsystem.R
+import java.util.Locale
+import kotlinx.datetime.Clock
+import kotlinx.datetime.LocalDate
+
+@Composable
+fun TaskyDatePicker(
+    selectedDate: LocalDate,
+    onSelectDate: (LocalDate) -> Unit,
+    expanded: Boolean,
+    toggleExpanded: () -> Unit,
+    modifier: Modifier = Modifier,
+    clock: Clock = Clock.System,
+    content: @Composable () -> Unit,
+) {
+    // don't use rememberDatePickerState because we want to be able to set the selected date
+    // from the date widget on the agenda overview screen too
+    val datePickerState = DatePickerState(
+        locale = Locale.getDefault(),
+        initialSelectedDateMillis = selectedDate.toMillis(),
+    )
```

Comment:
I still don't fully understand how `rememberDatePickerState` prevents you from doing that 🤔 
If you open the date picker, you pass in the initial date to show there, but when would you need to change the date of the opened date picker from outside the picker? And even if you'd need to, could it be a solution to let the calling composable pass this date picker state as a parameter (with remember), so it could use it to change the date?

Reply:
I think I got it sorted, please check out https://github.com/evanadwyer-pronoid-software/Tasky/pull/70 🙂 I was overthinking it 😂